src/toplevel.rs

```diff
@@ -236,6 +222,7 @@ impl<ErrType: ErrTypeTraits> Toplevel<ErrType> {
     ///
     /// An error of type [`GracefulShutdownError`] if an error occurred.
     ///
+    #[tracing::instrument(name = "Global System Shutdown", skip_all, level = "debug", fields(timeout = %humantime::format_duration(shutdown_timeout)))]
```

Comment:
Would you mind explaining the rationale for this line? Is this best practice to add a `name` whenever possible? I kind of find it distracting. Either way, `Global System Shutdown` is kind of incorrect, because people could use nested toplevels and similar. I would prefer to leave the name out, as it will only lead to discussion concerning the correctness of the name. Can't we simply leave out the entire `#[tracing::instrument()]` line? Why do we have it? What's wrong with the default? Maybe I simply don't know enough about `tracing`.

Reply:
After reading a little more, I now kind of understand the concept of spans and grasp what `#[tracing::instrument]` is for. I don't quite understand what we need the `humantime` crate for, as `Duration` is already `Debug` printable. Further, I don't understand why `skip_all`, followed by `fields` if we could simply do `skip(self)`.
Lastly, I kind of don't quite understand why we need this in general; this will never enter user code, so what's the point in using spans here?

Reply:
Further, why do we need any of this if there is already [`tracing-log`](https://docs.rs/tracing-log/0.1.3/tracing_log/#convert-log-records-to-tracing-events)? What's the advantage of switching to `tracing` in this crate?

Reply:
tracing-log is just convert every log into event, if we use many layers of subsystems, it's hard to tell which message is from where.

Reply:
I understand, but the subsystems themselves should then derive `#[tracing::instrument]`, and there is nothing in this crate that prevents this. Setting a span within our code won't do anything for the subsystems themselves, as they are not connected to our code, they get spawned via `tokio::spawn`.

Reply:
You're right, sorry, I didn't think well.

Reply:
I did think about moving over to tracing either ay at some point. The reason I didn't so far was compatibility - it's a lot easier to use this crate in `tracing` if it itself implements `log` than the other way round. But of course I could be mistaken; if it is equally simple to use this crate in `log` if itself implements `tracing`, then I'm actually more inclined to move over to `tracing` than not :) After playing around with `bevy` I do think that `tracing` is the future.

Reply:
Either way, regardless of whether we move the crate itself over, I definitely would like to move the examples over to `tracing` to demonstrate how it should work. But if both can happen simultaneously without breaking too much, I think we should continue with this branch. Don't be intimidated by my feedback, I tend to question things. Not to discredit them, but to understand them :)

Reply:
If you don't want to continue because it no longer is a valid usecase for you, I might take over your branch if that's fine?

Reply:
> If you don't want to continue because it no longer is a valid usecase for you, I might take over your branch if that's fine?

OK